app/index.js

```diff
@@ -1,16 +1,24 @@
 import React from 'react'
 import ReactDOM from 'react-dom'
+import {Router, Route, IndexRoute, hashHistory} from 'react-router';
 
 import App from './components/App'
 import ProductList from './components/ProductList'
+import ProductDetail from './components/ProductList/productDetail'
+
+import products from './products.json';
 
 import './index.css'
 
 ReactDOM.render(
-
-  <App>
-    <ProductList />
-  </App>, 
+      <Router history={hashHistory}>
+        <Route component={App}>
+          <Route path="/"  component={ProductList} data={products} />
+          <Route path="/product/:id" component={ProductDetail}  data={products} />
```

Comment:
It seems unnecessary to pass all products to the detail page. It shouldn't be the responsibility of the detail page to find the correct item. But in the current setup I don't know a better way. This will be better once you use redux or some other state management solution